tests/memory_checker/test_memory_checker.py

```diff
@@ -165,47 +290,109 @@ def consume_memory_and_restart_container(duthost, container_name, vm_workers, lo
+    loganalyzer.expect_regex = []
+    loganalyzer.expect_regex.extend(expected_alerting_messages)
+    marker = loganalyzer.init()
+
+    thread_pool = ThreadPool()
+    thread_pool.apply_async(consume_memory, (duthost, container_name, vm_workers))
+
+    logger.info("Sleep 130 seconds to wait for the alerting messages from syslog ...")
+    time.sleep(130)
+
+    logger.info("Checking the alerting messages related to container restart ...")
+    loganalyzer.analyze(marker)
+    logger.info("Found all the expected alerting messages from syslog!")
+
+    logger.info("Waiting for '{}' container to be restarted ...".format(container_name))
+    restarted = wait_until(CONTAINER_RESTART_THRESHOLD_SECS,
+                           CONTAINER_CHECK_INTERVAL_SECS,
+                           0,
+                           check_container_state, duthost, container_name, True)
+    pytest_assert(restarted, "Failed to restart '{}' container!".format(container_name))
+    logger.info("'{}' container is restarted.".format(container_name))
 
-def test_memory_checker(duthosts, enum_dut_feature_container, creds, enum_rand_one_per_hwsku_frontend_hostname):
-    """Checks whether the telemetry container can be restarted or not if the memory
-    usage of it is beyond the threshold. The `stress` utility is leveraged as
-    the memory stressing tool.
+    logger.info("Running 'stress' utility again in '{}' ...".format(container_name))
+    thread_pool.apply_async(consume_memory, (duthost, container_name, vm_workers))
+
+    check_counter = 0
+    marker = loganalyzer.update_marker_prefix("test_monit_counter")
+    logger.info("Checking memory usage of '{}' every 30 seconds for 6 times ...".format(container_name))
+    while check_counter < 6:
+        check_counter += 1
+        mem_usage = get_container_mem_usage(duthost, container_name)
+        logger.info("Memory usage of '{}' is '{}'".format(container_name, mem_usage))
+        time.sleep(30)
+
+    logger.info("Analyzing syslog messages to verify whether '{}' is restarted ...".format(container_name))
+    analyzing_result = loganalyzer.analyze(marker, fail=False)
+    if not new_syntax_enabled:
+        pytest_assert(analyzing_result["total"]["expected_match"] == 0,
+                      "Monit can reset counter and restart '{}'!".format(container_name))
+        logger.info("Monit was unable to reset its counter and '{}' can not be restarted!".format(container_name))
+    else:
+        pytest_assert(analyzing_result["total"]["expected_match"] == 5,
+                      "Monit still can not restart '{}' with the help of new syntax!".format(container_name))
+        logger.info("Monit was able to restart '{}' with the help of new syntax!".format(container_name))
+
+
+@pytest.mark.parametrize("test_setup_and_cleanup",
+                         ['    if status == 3 for 1 times within 2 cycles then exec "/usr/bin/restart_service telemetry"'],
```

Comment:
>status [](http://example.com/codeflow?start=34&length=6)

Provide the reference link for `status` values?

Reply:
Link is at here: https://github.com/Azure/sonic-buildimage/blob/122cb9069598e0798883538af923381c40f3ba8a/files/image_config/monit/memory_checker#L88